Tests/UnitTests/StoreKitExtensions/SKErrorTests.swift

```diff
@@ -0,0 +1,148 @@
+    }
+
+    func testPaymentInvalid() {
+        let error: SKError = .init(.paymentInvalid)
+
+        verifyPurchasesError(error,
+                             expectedCode: .purchaseInvalidError,
+                             underlyingError: error)
+    }
+
+    func testProductNotAvailableError() {
+        let error: SKError = .init(.storeProductNotAvailable)
+
+        verifyPurchasesError(error,
+                             expectedCode: .productNotAvailableForPurchaseError,
+                             underlyingError: error)
+    }
+
+    @available(iOS 14.0, macOS 11.0, watchOS 6.2, *)
+    @available(tvOS, unavailable)
+    func testUnsupportedPlatformError() throws {
+        try AvailabilityChecks.iOS14APIAvailableOrSkipTest()
+
+        let error: SKError = .init(.unsupportedPlatform)
+
+        verifyPurchasesError(error,
+                             expectedCode: .purchaseNotAllowedError,
+                             underlyingError: error)
+    }
+
+    @available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 6.2, *)
+    func testIneligibleForOfferError() throws {
+        try AvailabilityChecks.iOS14APIAvailableOrSkipTest()
+
+        let error: SKError = .init(.ineligibleForOffer)
+
+        verifyPurchasesError(error,
+                             expectedCode: .ineligibleError,
+                             underlyingError: error)
+    }
+
+    @available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 6.2, *)
+    func testInvalidOfferError() throws {
+        try AvailabilityChecks.iOS14APIAvailableOrSkipTest()
+
+        let error: SKError = .init(.invalidOfferIdentifier)
+
+        verifyPurchasesError(error,
+                             expectedCode: .invalidPromotionalOfferError,
+                             underlyingError: error)
+    }
+
+    func testCurrentlySubscribedError() throws {
+        let error = try XCTUnwrap(
+            NSError(
+                domain: SKErrorDomain,
+                code: SKError.unknown.rawValue,
+                userInfo: [
+                    NSUnderlyingErrorKey: NSError(
+                        domain: "ASDServerErrorDomain",
+                        code: 3532,
+                        userInfo: [:]
+                    )
+                ]
+            ) as? SKError
+        )
+
+        verifyPurchasesError(error,
+                             expectedCode: .productAlreadyPurchasedError,
+                             underlyingError: error)
+    }
+
+    func testPaymentSheetCancelledError() throws {
+        let error = try XCTUnwrap(
+            NSError(
+                domain: SKErrorDomain,
+                code: 907,
+                userInfo: [
+                    NSUnderlyingErrorKey: NSError(
+                        domain: "AMSErrorDomain",
+                        code: 6,
+                        userInfo: [:]
+                    )
+                ]
+            ) as? SKError
+        )
+
+        verifyPurchasesError(error,
```

Comment:
This is so good that we have tests for this. I'm loving the new errors

Reply:
Yeah I'm using this new helper in all of these which makes it very simple :)
<img width="619" alt="Screen Shot 2022-04-11 at 10 25 35" src="https://user-images.githubusercontent.com/685609/162795824-6837c709-bfc1-4ba0-b46a-d0c3572ffbaa.png">